src/main/java/listeners/MainEventListener.java

```diff
@@ -0,0 +1,232 @@
+        String[] tokens = event.getMessage().getContentDisplay().split(" ");
+        if (!tokens[0].startsWith("$")) return;
+
+        String guildID = event.getGuild().getId();
+        Server server = servers.computeIfAbsent(guildID, k -> Server.load(event.getGuild()));
+
+        // pick the correct method to call
+        CommandHandler commandHandler;
+        switch (tokens[0].substring(1)) {
+            case "help" -> commandHandler = this::help;
+            case "maketopic" -> commandHandler = this::maketopic;
+            case "deletetopic" -> commandHandler = this::deletetopic;
+            case "showtopics" -> commandHandler = this::showtopics;
+            case "queue" -> commandHandler = this::queue;
+            case "ready" -> commandHandler = this::ready;
+            case "showqueue" -> commandHandler = this::showqueue;
+            case "clear" -> commandHandler = this::clear;
+            default -> commandHandler = this::unknownCommand;
+        }
+        commandHandler.handle(
+                Objects.requireNonNull(event.getMember()),
+                event.getChannel(),
+                server,
+                Arrays.copyOfRange(tokens, 1, tokens.length));
+    }
+
+    private void help(Member member, TextChannel channel, Server server, String[] args) {
+        EmbedBuilder embedBuilder = new EmbedBuilder();
+        embedBuilder.setTitle("Help!");
+        embedBuilder.setDescription("Possible commands:");
+        embedBuilder.setColor(0xE57D25);
+
+        embedBuilder.addField("$queue <topic>", "Add yourself to a queue.", false);
+        embedBuilder.addField("$showqueue <topic>", "Show the people currently in queue.", false);
+        embedBuilder.addField("$showtopics", "List all topics.", false);
+        embedBuilder.addField("$ready <topic> (admin only)", "Retrieve the next person from the queue.", false);
+        embedBuilder.addField("$clear <topic> (admin only)", "Clear the specified queue.", false);
+        embedBuilder.addField("$maketopic <name> (admin only)", "Create a new topic.", false);
+        embedBuilder.addField("$deletetopic <name> (admin only)", "Delete a topic.", false);
+
+        channel.sendMessage(embedBuilder.build()).queue();
+    }
+
+    private void maketopic(Member member, TextChannel channel, Server server, String[] args) {
+        if (!checkAdmin(member, channel)) return;
+
+        Topic topic = new Topic(args[0]);
+        server.createTopic(topic);
+
+        channel.sendMessage(String.format(
+                "%s Topic role \"%s\" has been created.",
+                member.getAsMention(),
+                args[0])).queue();
+    }
+
+    private void deletetopic(Member member, TextChannel channel, Server server, String[] args) {
+        if (!checkAdmin(member, channel)) return;
+
+        Arrays.stream(server.getTopics())                   // loop over all topics...
+                .filter(t -> t.getName().equals(args[0]))   // ...if this topic's name is args[0]...
```

Comment:
What is `args[0]`?

Reply:
`args[0]` is the first parameter provided to the command. For example, if a user calls `$deletetopic Java`, `args == ["Java"]`.

Reply:
I ask because I known it's the first arg of the command, but _semantically_? Like, is it the `channelToDelete`, that kinda thing

Reply:
I can add an extra variable to make it read better semantically. Something like `string topicName = args[0];` on line 134?

Reply:
That'd be great! I just think it helps the code clarity

Reply:
Added the extra variables where necessary